### DISCUSSION TITLE
Combining two packages: openradioss-starter and openradioss-engine

### DISCUSSION THREAD

**Discussion by laurent-altr** [spack/spack#43446] [General]
Hello,

I am one of the maintainers of OpenRadioss, and I am new to Spack. 
OpenRadioss is available via two packages: openradioss-starter and openradioss-engine. The starter binary handles modeling error checks and mesh splitting, while the engine binary is responsible for numerical simulations. 

My concern is regarding compatibility. The binaries from different commits may not work together. Considering this, would it be feasible to gather these into a single Spack package? 

Currently, they are built in separate subdirectories of our git repository, using two individual CMake projects. Is this the reason for the two distinct packages?

Thanks

**Comment by alalazo** [comment]
@kjrstory You're listed as a maintainer for both packages, do you have any concern with the proposal above?

**Comment by kjrstory** [comment]
I had many concerns when I initially created the packages. I am well aware that the starter and engine are used in pairs, but these two binaries have different build files and dependencies.
Due to the rarity of managing two builds within Spack, I was compelled to separate them into two packages.
Another difficulty is that OpenRadioss does not have specific releases. Therefore, in Spack, it always builds from the latest commit on the main branch. To address this, we can arbitrarily set specific versions in Spack.
For example, we could designate the commit from April 16 as a specific version, and then make sure that when building starter and engine, they are built with each other's versions.

If installed like this:
```
spack install openradioss-starter@20240416
```
Then, if you run:
```
spack install openradioss-engine
```

It must install the 20240416 version as well, even if "@20240416" is not specified.
What do you think?
